src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -294,24 +294,27 @@ public virtual VerifyResult VerifyForEachBlock(StoreView snapshot, TransactionVe
             foreach (TransactionAttribute attribute in Attributes)
                 if (!attribute.Verify(snapshot, this))
                     return VerifyResult.Invalid;
-            if (hashes.Length != Witnesses.Length) return VerifyResult.Invalid;
-            for (int i = 0; i < hashes.Length; i++)
-            {
-                if (Witnesses[i].VerificationScript.Length > 0) continue;
-                if (snapshot.Contracts.TryGet(hashes[i]) is null) return VerifyResult.Invalid;
-            }
+            long net_fee = NetworkFee - Size * NativeContract.Policy.GetFeePerByte(snapshot);
+            if (!this.VerifyWitnesses(snapshot, net_fee, WitnessFlag.NonStandardWitness))
```

Comment:
All `Verification`s triggers are state independent. Because they have `CallFlags.None`.

Reply:
`CallFlags.None` means it has not been initialized. `CallFlags.StandardWitness` is state independent and `CallFlags.NonStandardWitness` is state dependent.

Reply:
I say `CallFlags`. Not `WitnessFlag`. We don't need `WitnessFlag`. Because all `Verification` triggers are state independent.

Reply:
> All Verification triggers are state independent. 

If so, we should modify `System.Contract.IsStandard`, as it's state  dependent. https://github.com/neo-project/neo/blob/4a61eabad7bbeeee115666df3ea65640078c8fac/src/neo/SmartContract/ApplicationEngine.Contract.cs#L23

Reply:
See #1830

Reply:
I redefined WitnessFlag so that Witnesses are splitted into 2 parts, state independent or state dependent depending on the length of VerificationScript. 2 kinds of witnesses are checked in `Transaction.VerifyStateIndependent` and `Transaction.VerifyStateDependent` respectively